app/models/accounting/transaction.rb

```diff
@@ -246,7 +246,8 @@ def net_debit_for_account(account_id)
   end
 
   def respect_closed_books_date
-    return if division.closed_books_date.nil? || txn_date > division.closed_books_date
-    errors.add(:txn_date, :closed_books_date, date: division.closed_books_date)
+    cbd = Loan.find(project_id).qb_division.closed_books_date
```

Comment:
why are we doing the explicit find here?

Reply:
closed_books_date was coming back nil if I just use `loan`. I don't know why but I assume its because its in the validation step and the record hasn't been saved yet. The project_id attr is available, but ActiveRecord associations don't seem to be.

Reply:
there should be a way to do this without the explicit find but i don't have the setup to test this locally right now to validate my theory so this is fine, though a TODO: comment reflecting we should try removing that would be good just so that doesn't get lost until i've got time to muck around with it

Reply:
@meltheadorable I tried this again and I am able to use the association, so I'm doing that. I must have gotten mixed up when I was testing. Thanks for catching this!